### PR TITLE
Add InMemoryCommitStore to test Managed Commit Backend

### PR DESCRIPTION
**Pull request by sumeet-db**
#### Which Delta project/connector is this regarding?

- [x] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [ ] Other (fill in here)

## Description

An in-memory-commit-store that tracks per-table commits, backfills and validates various edge cases and unexpected scenarios.

## How was this patch tested?

Unit tests

## Does this PR introduce _any_ user-facing changes?

No

### CODE REVIEW

File: spark/src/main/scala/org/apache/spark/sql/delta/managedcommit/AbstractBatchBackfillingCommitStore.scala [hunk 1]
```diff
@@ -119,17 +121,27 @@ trait AbstractBatchBackfillingCommitStore extends CommitStore with Logging {
...
     } catch {
       case _: FileAlreadyExistsException =>
         logInfo(s"The backfilled file $targetFile already exists.")
+    } finally {
+      commitContentIterator.close()
     }
   }
 
+  /** Callback to tell the CommitStore that all commits <= `untilVersion` are backfilled. */
+  protected[managedcommit] def registerBackfill(
```
Comment (prakharjain09) [inline spark/src/main/scala/org/apache/spark/sql/delta/managedcommit/AbstractBatchBackfillingCommitStore.scala:141]: Why is this `protected[managedcommit]` and not protected like other APIs?

This class could be extended by CommitStores even outside this module.

**Review by prakharjain09 — APPROVED** [review]

File: spark/src/main/scala/org/apache/spark/sql/delta/managedcommit/AbstractBatchBackfillingCommitStore.scala (hunk 1 above)
Comment (sumeet-db) [reply spark/src/main/scala/org/apache/spark/sql/delta/managedcommit/AbstractBatchBackfillingCommitStore.scala]: Done

File: spark/src/main/scala/org/apache/spark/sql/delta/managedcommit/InMemoryCommitStore.scala [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+      val tableData = perTableMap.get(tablePath)
+      // Calculate the end version for the range, or use the last key if endVersion is not provided
+      val effectiveEndVersion =
+        endVersion.getOrElse(tableData.commitsMap.lastOption.map(_._1).getOrElse(startVersion))
+      val commitsInRange = tableData.commitsMap.range(startVersion, effectiveEndVersion + 1)
+      commitsInRange.values.toSeq
+    }
+  }
+
+  override protected[managedcommit] def registerBackfill(
```
Comment (prakharjain09) [inline spark/src/main/scala/org/apache/spark/sql/delta/managedcommit/InMemoryCommitStore.scala:116]: This also needs to be fixed?

Comment (sumeet-db) [reply spark/src/main/scala/org/apache/spark/sql/delta/managedcommit/InMemoryCommitStore.scala]: Done

**Pull request closed without merging**